// coding: utf-8
//
// Copyright 2020 The Oppia Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS-IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "core/domain/proto/text_classifier.proto";

// Training job response payload contains job result of the training job
// along with other metadata items such as vm_id (to identify which VM executed
// this job) and signature of the payload for security purpose.
message TrainingJobResponsePayload {
  // Job result of the training job. Job result contains the ID of the Job and
  // trained model (frozen model) of the job.
  message JobResult {
    // Id of the training job whose data is being stored.
    string job_id = 1;
    
    // Each of the classifier algorithms' proto message must be present in
    // the oneof classifier_data field.
    oneof classifier_frozen_model {
      TextClassifierFrozenModel text_classifier = 2;
    }
  }
  JobResult job_result = 1;

  // Id of the VM instance that trained the job.
  string vm_id = 2;
  
  // Signature of the job data for authenticated communication.
  string signature = 3;
}
